### ISSUE TITLE
save_json = True parameter makes trainig slower

### ISSUE THREAD

**Issue by rorro6787** [ultralytics/ultralytics#16775]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I am training the yolov8n-seg.pt model for a segmentation tasks and I have noticed that when I use the hiperparameter save_json=True for the training process, not only it gets extremely slow but, sooner or later my vsc crushes and I have repeat the training. It only happens when I have that parameter at True. Any reason?

### Environment

Ultralytics 8.3.3 🚀 Python-3.12.3 torch-2.3.1+cu121 CUDA:0 (NVIDIA GeForce RTX 2070 with Max-Q Design, 7967MiB)
Setup complete ✅ (12 CPUs, 15.4 GB RAM, 475.8/936.8 GB disk)
None

### Minimal Reproducible Example

```python
# Load the pre-trained YOLOv8n model for segmentation
model = YOLO('yolov8n-seg.pt', task='segmentation')  

save_directory = os.path.join(path, 'runs')

if not os.path.exists(save_directory):
      os.makedirs(save_directory)

  # Train the model with your dataset
  model.train(
      data=yaml_file_path,       # Path to the YAML file with data configuration
      epochs=32,                 # Number of training epochs
      imgsz=320,                 # Image size (width and height)
      # lr0=0.01,                # Initial learning rate
      # lrf=0.001,               # Final learning rate
      # weight_decay=0.0005,     # Weight decay for regularization
      # momentum=0.937,          # Momentum for SGD optimizer
      # dampening=0.5,           # Momentum damping
      # nesterov=True,           # Use Nesterov momentum
      # accumulative=2,          # Gradient accumulation steps
      batch=-1,                  # Batch size, -1 for default
      name=name_model,           # Experiment/model name
      device=0,                  # Device ID for training (0 for first GPU)
      project=save_directory,    # Project directory for results
      save_dir=save_directory,   # Directory to save the trained model
      fraction=0.25,             # Fraction of dataset for training
      # hyp=None,                # Hyperparameter file path or None for defaults
      # local_rank=-1,           # Local GPU rank for distributed training
      # sync_bn=False,           # Use synchronized batch norm
      # workers=8,               # Number of data loading workers
      plots=True,                # Generate training plots
      # freeze=[0],              # Freeze specific layers (list of layer indices)
      # save_period=1,           # Save model every 'n' epochs
      # resume=False,            # Resume training from a saved model
      # val=True,                # Validate model after each epoch
      # image_weights=False,     # Weight images in loss
      # hyp_path=None,           # Path to hyperparameter file
      # save_json=True,          # Save results as JSON
      # lr_schedule=True,        # Use learning rate scheduling
      # rect=False,              # Use rectangular image resizing
      # single_cls=False,        # Train on single class only
      # compute_map=False,       # Calculate mAP during validation
  )
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @rorro6787, thank you for reaching out and sharing your concern with Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will assist you soon.

From your description, it seems you're experiencing a slowdown with the `save_json=True` parameter during training. We recommend checking our [documentation](https://docs.ultralytics.com) for common setup guidance.

### For 🐛 Bug Reports

Please ensure you've provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This helps us pinpoint the issue more efficiently.

### Helpful Resources

- **Upgrade** to the latest package to see if the issue persists:
  ```bash
  pip install -U ultralytics
  ```

- Check for compatibility with Python 3.12 and your current PyTorch version.

- Engage with our community on [Discord](https://ultralytics.com/discord) 🎧 for real-time discussion, or explore detailed topics on [Discourse](https://community.ultralytics.com).

### Running Environments

Make sure to run your experiments in one of the verified environments:

- **Notebooks** with free GPU: 
  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> 
  <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> 
  <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)

- **AWS**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)

- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### CI Status

If the [CI Badge](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) is green, all tests are passing which indicates no known issues on the master branch.

Feel free to update your issue with any additional details or questions. We appreciate your patience and contribution to improving our tools! 😊

**Comment by rorro6787** [comment]
@ambitious-octopus but the bigger your dataset, it get exponentially bigger the time it takes for the system to validate when using save_json. I have the same problem if I add it as a validatation argument instead of training. It is not that the program crushes, I guess my GPU cannot handle so long time validating and just closes VSC

**Comment by Laughing-q** [comment]
@rorro6787 Hey the validation with `save_json=True` could be very slow especially with segmentation training, I'll suggest you to set `save_json=False` for training if you don't have a specific reason using it. :)
